sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowGroupLimitExec.scala

```diff
@@ -116,7 +129,8 @@ abstract class BaseLimitIterator extends Iterator[InternalRow] {
 
 case class SimpleLimitIterator(
     input: Iterator[InternalRow],
-    limit: Int) extends BaseLimitIterator {
+    limit: Int,
+    numOutputRows: SQLMetric) extends BaseLimitIterator {
```

Comment:
Could we avoid it as parameter ?
It seems just put
```
override lazy val metrics = Map(
     "numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"))
```
into BaseLimitIterator directly.

Reply:
hmm.. are you sure it works ?

Reply:
I have checked before comment..

Reply:
`override lazy val metrics` must be put in a `SparkPlan`

Reply:
I got it.

Reply:
@ulysses-you I'm sorry. Because metrics must be put in a `SparkPlan`, it seems the former code is better.

Reply:
nvm, it's more similar with existed code `SortMergeJoinExec` which also pass metric to iterator